### PR TITLE
Perf optimization for mailing list

### PR DESCRIPTION
**Pull request by tgxworld**

### CODE REVIEW

**Comment by coding-horror** [comment]
Why does this need to be a PR?

File: lib/email/styles.rb [hunk 1]
```diff
@@ -228,7 +228,7 @@ def replace_relative_urls
 
       @fragment.css('[href]').each do |element|
         href = element['href']
-        if href =~ /^\/\/#{host}/
+        if href.start_with?("\/\/#{host}")
```
Comment (ZogStriP) [inline lib/email/styles.rb]: I'm curious, what's the perf improvement of using `start_with?` as opposed to the regex?

File: app/mailers/user_notifications.rb [hunk 2]
```diff
@@ -80,10 +80,21 @@ def mailing_list(user, opts={})
     @since = opts[:since] || 1.day.ago
     @since_formatted = short_date(@since)
 
-    @new_topic_posts      = Post.mailing_list_new_topics(user, @since).group_by(&:topic) || {}
```
Comment (ZogStriP) [inline app/mailers/user_notifications.rb]: Do we use these scope elsewhere in the app? If not they should probably be removed.

**Comment by ZogStriP** [comment]
LGTM 👍

**Comment by tgxworld** [comment]
> Why does this need to be a PR?

I'm just making sure I don't break the tests

**Comment by ZogStriP** [comment]
> I'm just making sure I don't break the tests

What don't you run the specs on your machine?

File: lib/email/styles.rb (hunk 1 above)
Comment (tgxworld) [inline lib/email/styles.rb]: `start_with?` is faster https://github.com/JuanitoFatas/fast-ruby#stringmatch-vs-stringstart_withstringend_with-code-start-code-end

I was curious as well so I wrote this script

```
require 'memory_profiler'

test_string = 'abcde'
abc = "abc"
MemoryProfiler.report do
  test_string.start_with?(abc)
  # test_string =~ /\/\/#{abc}/
end.pretty_print
```

`start_with?` doesn't allocate any objects while using Regexp allocates 4 strings for this case.

**Pull request closed without merging**